### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by JakovTomasic**
I implemented a few fixes and improvements. Each commit is a separate fix explained in the commit's message.

### CODE REVIEW

File: client/src/components/SearchForm.tsx
[the remark this answers is not in the document]
Comment (JakovTomasic) [reply client/src/components/SearchForm.tsx:1]: Pull changes and try again. The commit 4466edb35e53413562f4951ec905cdd74b67fe25 maybe fixed it.

I don't have the bug you mentioned. I did have it before, but I fixed it before opening this pull request.

**Comment by JakovTomasic** [comment]
Please review the new changes, too.

I fixed that weird two nodes that were too close in fe8332f821a0b6d0f2bd97a78399308a83f7bec1.

File: client/src/components/SearchForm.tsx
[the remark this answers is not in the document]
Comment (Marko132001) [reply client/src/components/SearchForm.tsx:1]: This doesn't fix it for me. In some cases it works. For example, if you define as current location the "cantine" and as destination room "1-117a". But if you define for example, for current location "cantine" and for destination room "P4" the behaviour is different. Try to test it a little more if you haven't already. If this poses as a bigger problem we can try to solve it in a different issue.

**Review by Marko132001 — CHANGES_REQUESTED** [review]
The switch button issue still does not work for me. The problematic node fix works great now. All in all, try to solve the switch button problem, but as far as I'm concerned we can postpone it for later, up to you.

**Review by Marko132001 — APPROVED** [review]
Everything works now. You can merge.

**Pull request merged**